CAM_interface/yog_intr.F90

```diff
@@ -0,0 +1,232 @@
+module yog_intr
```

Comment:
not sure this is the best name, as it's not immediately clear to me what "intr" stands for, and therefore what this module is

Reply:
This matches the naming system for CAM physics modules so I would be inclined to stay consistent.
e.g. clubb_intr.F90, carma_intr.F90
See - https://github.com/m2lines/CAM-ML/tree/CAM-ML/src/physics/cam

Reply:
There is a docstring below to add context: https://github.com/m2lines/convection-parameterization-in-CAM/blob/a515164a5ce32d49ed247f7096e566730b33d5fb/CAM_interface/yog_intr.F90#L1-L10